app/app.rb

```diff
@@ -0,0 +1,66 @@
+ENV["RACK_ENV"] ||=  "development"
+require 'sinatra/base'
+require "sinatra/flash"
+require_relative 'data_mapper_setup'
+require_relative "models/peep"
+require_relative "models/user"
+
+class Chitter < Sinatra::Base
+  use Rack::MethodOverride
+  register Sinatra::Flash
+  enable :sessions
+  set :session_secret, 'super secret'
+
+  get '/' do
+    erb :index
+  end
+
+  post '/log_in' do
+    user = User.authenticate(params[:username], params[:password])
+   if user
```

Comment:
Watch out for this indentation! Have a look at a Ruby style guide for idiomatic Ruby indentation.